system/Commands/Generators/CreateController.php

```diff
@@ -0,0 +1,168 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @package    CodeIgniter
+ * @author     CodeIgniter Dev Team
+ * @copyright  2019-2020 CodeIgniter Foundation
+ * @license    https://opensource.org/licenses/MIT	MIT License
+ * @link       https://codeigniter.com
+ * @since      Version 4.0.0
+ * @filesource
+ */
+
+namespace CodeIgniter\Commands\Generators;
+
+use CodeIgniter\CLI\BaseCommand;
+use CodeIgniter\CLI\CLI;
+use Config\Services;
+
+/**
+ * Creates a new controller file.
+ *
+ * @package CodeIgniter\Commands
+ */
+class CreateController extends BaseCommand
+{
+	/**
+	 * The group the command is lumped under when listing commands.
+	 *
+	 * @var String
+	 */
+	protected $group = 'Generators';
+
+	/**
+	 * The Command's name
+	 *
+	 * @var String
+	 */
+	protected $name = 'make:controller';
+
+	/**
+	 * The Command's short description
+	 *
+	 * @var String
+	 */
+	protected $description = 'Creates a new controller file.';
+
+	/**
+	 * The Command's usage
+	 *
+	 * @var String
+	 */
+	protected $usage = 'make:controller [name] [options]';
+
+	/**
+	 * The Command's arguments
+	 *
+	 * @var Array
+	 */
+	protected $arguments = [
+		'name' => 'The controller file name',
+	];
+
+	/**
+	 * The Command's options
+	 *
+	 * @var Array
+	 */
+	protected $options = [
+		'-n' => 'Set controller namespace',
+	];
+
+	/**
+	 * Creates a new controller file.
+	 *
+	 * @param Array $params
+	 */
+	public function run(Array $params): void
+	{
+		helper('inflector');
+
+		$name = array_shift($params);
+
+		if (empty($name))
+		{
+			$name = CLI::prompt(lang('Controller.nameFile'), null, 'required');
+		}
+
+		$ns       = $params['-n'] ?? CLI::getOption('n');
+		$homepath = APPPATH;
+
+		if (! empty($ns))
+		{
+			// Get all namespaces
+			$namespaces = Services::autoloader()->getNamespace();
+
+			foreach ($namespaces as $namespace => $path)
+			{
+				if ($namespace === $ns)
+				{
+					$homepath = realpath(reset($path)) . DIRECTORY_SEPARATOR;
+					break;
+				}
+			}
+		}
+		else
+		{
+			$ns = defined('APP_NAMESPACE') ? APP_NAMESPACE : 'App';
+		}
+
+		// Full path
+		$path = $homepath . 'Controllers/' . $name . '.php';
+
+		// Class name should be pascal case now (camel case with upper first letter)
+		$name = pascalize($name);
+
+		$template = <<<EOD
+		<?php namespace $ns\Controllers;
+
```

Comment:
Maybe add `use App\Controllers\BaseController` in case their namespace isn't the default, otherwise they will have to have a `BaseController` class in their namespace, which seems counter-intuitive.